packages/synapse-react-client/src/components/AclEditor/useUpdateAcl.ts

```diff
@@ -35,57 +40,64 @@ export default function useUpdateAcl(options: UseUpdateAclOptions) {
     onChange(resourceAccessList)
   }, [resourceAccessList])
 
-  const addResourceAccessItem = (
-    newReviewerId: string | null,
-    accessTypes: ACCESS_TYPE[],
-  ) => {
-    setIsDirty(true)
-    if (newReviewerId) {
-      const alreadyReviewer = resourceAccessList.some(
-        resourceAccess => resourceAccess.principalId === Number(newReviewerId),
-      )
-      if (alreadyReviewer) {
-        onError(PRINCIPAL_ALREADY_ADDED_ERROR_MESSAGE)
-      } else {
-        const newResourceAccess: ResourceAccess = {
-          principalId: Number(newReviewerId),
-          accessType: accessTypes,
-        }
-        const updatedResourceAccessList = [
-          ...resourceAccessList,
-          newResourceAccess,
-        ]
-        setResourceAccessList(updatedResourceAccessList)
+  const addResourceAccessItem = useCallback(
+    (newReviewerId: string | null, accessTypes: ACCESS_TYPE[]) => {
+      setIsDirty(true)
+      if (newReviewerId) {
+        setResourceAccessList(resourceAccessList => {
```

Comment:
use setState callback for all of these so consecutive calls are properly batched